kyuubi-util-scala/src/main/scala/org/apache/kyuubi/util/reflect/ReflectUtils.scala

```diff
@@ -37,32 +37,47 @@ object ReflectUtils {
       DynClasses.builder().loader(cl).impl(className).buildChecked()
     }.isSuccess
 
-  def getFieldVal[T](target: Any, fieldName: String): T =
+  def getField[T](target: Any, fieldName: String): T = {
+    val targetClass = target.getClass
     Try {
-      DynFields.builder().hiddenImpl(target.getClass, fieldName).build[T]().get(target)
+      DynFields.builder().hiddenImpl(targetClass, fieldName)
+        .buildChecked[T](target).get()
     } match {
       case Success(value) => value
-      case Failure(e) =>
-        val candidates = target.getClass.getDeclaredFields.map(_.getName).mkString("[", ",", "]")
-        throw new RuntimeException(s"$fieldName not in ${target.getClass} $candidates", e)
+      case Failure(t) =>
+        val candidates = targetClass.getFields.map(_.getName).sorted
+        throw new RuntimeException(
+          s"Field $fieldName not in $targetClass [${candidates.mkString(",")}]",
+          t)
     }
+  }
 
-  def getFieldValOpt[T](target: Any, name: String): Option[T] =
-    Try(getFieldVal[T](target, name)).toOption
-
-  def invoke(target: AnyRef, methodName: String, args: (Class[_], AnyRef)*): AnyRef =
-    try {
-      val (types, values) = args.unzip
-      DynMethods.builder(methodName).hiddenImpl(target.getClass, types: _*).build()
-        .invoke(target, values: _*)
-    } catch {
-      case e: NoSuchMethodException =>
-        val candidates = target.getClass.getMethods.map(_.getName).mkString("[", ",", "]")
-        throw new RuntimeException(s"$methodName not in ${target.getClass} $candidates", e)
+  /**
+   * Invoke a method with the given name and arguments on the given target object.
+   * @param target the target object
+   * @param methodName the method name
+   * @param args pairs of class and values for the arguments
+   * @tparam T the expected return class type,
+   *           returning type Nothing if it's not provided or inferable
+   * @return
+   */
+  def invoke[T](target: AnyRef, methodName: String, args: (Class[_], AnyRef)*): T = {
```

Comment:
the `invokeAs` should be fine which also reduces the diff

Reply:
Adopted.